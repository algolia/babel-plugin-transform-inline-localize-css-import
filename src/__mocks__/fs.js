let files = {};

export function readFileSync(filepath) { return files[filepath]; }
export function __setFiles(newFiles) { files = newFiles; }
