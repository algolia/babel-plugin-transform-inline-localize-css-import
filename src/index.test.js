/* eslint-env jest */
import {join} from 'path';
import {transform} from 'babel-core';
import babelPlugin from './index.js';
import {__setFiles} from 'fs';

jest.mock('fs');

const babelOptions = {
  babelrc: false,
  plugins: [
    [babelPlugin, {
      localize: ({token, jsFileName}) => `ais-${jsFileName}__${token}`,
    }],
  ],
};

beforeEach(() => __setFiles({}));

const testCases = [{
  name: 'basic',
  code: 'import theme from "./style.css";',
  filename: join(__dirname, 'SearchBox.js'),
  files: {
    [join(__dirname, 'style.css')]: '.className {}',
  },
}, {
  name: 'different filename',
  code: 'import theme from "./style.css";',
  filename: join(__dirname, 'yo.js'),
  files: {
    [join(__dirname, 'style.css')]: '.className {}',
  },
}, {
  name: 'different JavaScript code',
  code: 'import theme from "./style.css"; a + a; import "module"',
  filename: join(__dirname, 'SearchBox.js'),
  files: {
    [join(__dirname, 'style.css')]: '.className {}',
  },
}, {
  name: 'multiple class names',
  code: 'import theme from "./style.css";',
  filename: join(__dirname, 'SearchBox.js'),
  files: {
    [join(__dirname, 'style.css')]: '.a .b, .c ~ .d, .d > .e {color:blue}',
  },
}, {
  name: 'keyframes rule',
  code: 'import theme from "./style.css";',
  filename: join(__dirname, 'SearchBox.js'),
  files: {
    [join(__dirname, 'style.css')]: '.className{animation-name: hide;color:blue}@keyframes hide {0%{opacity:0}};',
  },
}];

testCases.forEach(({filename, code, files, name, only}) => {
  const test = only ? fit : it;
  test(name, () => {
    __setFiles(files);

    expect(
      transform(
        code,
        {...babelOptions, filename}
      ).code
    ).toMatchSnapshot();
  });
});
