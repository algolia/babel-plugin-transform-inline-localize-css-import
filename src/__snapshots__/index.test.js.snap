exports[`test basic 1`] = `
"var theme = {
  code: \".Component__className {}\",
  classNames: {
    className: \"Component__className\"
  }
};"
`;

exports[`test different JavaScript code 1`] = `
"var theme = {
  code: \".Component__className {}\",
  classNames: {
    className: \"Component__className\"
  }
};
a + a;import \"module\";"
`;

exports[`test different cssFilename 1`] = `
"var theme = {
  code: \".CustomComponent__className {}\",
  classNames: {
    className: \"CustomComponent__className\"
  }
};"
`;

exports[`test keyframes rule 1`] = `
"var theme = {
  code: \".Component__className{animation-name: Component__hide;color:blue}@keyframes Component__hide {0%{opacity:0}};\",
  classNames: {
    className: \"Component__className\"
  }
};"
`;

exports[`test multiple class names 1`] = `
"var theme = {
  code: \".Component__a .Component__b, .Component__c ~ .Component__d, .Component__d > .Component__e {color:blue}\",
  classNames: {
    a: \"Component__a\",
    b: \"Component__b\",
    c: \"Component__c\",
    d: \"Component__d\",
    e: \"Component__e\"
  }
};"
`;
